src/handler/active_requests.rs

```diff
@@ -38,7 +38,12 @@ impl ActiveRequests {
         match self.active_requests_nonce_mapping.remove(nonce) {
             Some(node_address) => match self.active_requests_mapping.remove(&node_address) {
                 Some(request_call) => Some((node_address, request_call)),
-                None => None,
+                // If the matching request_call mapping doesn't exist, rollback remove_by_nonce op
+                None => {
+                    self.active_requests_nonce_mapping
```

Comment:
In this case, I think it means we have an entry in `active_requests_nonce_mapping` but not one in `active_requests_mapping`. 

This is the situation we are trying to avoid and probably shouldn't add anything back.  I think in some cases the original code would panic in this scenario. Perhaps here we only panic in debug. So maybe in this case we put
```rust
debug_unreachable!()
```

Reply:
```debug_unreachable!()``` results in an unsafe code block, replaced the code that should be dead code with ```unreachable``` for now